pagerduty/user.go

```diff
@@ -402,6 +467,28 @@ func (s *UserService) CreateNotificationRule(userID string, rule *NotificationRu
 	return v.NotificationRule, resp, nil
 }
 
+func (s *UserService) searchExistingNotificationRule(userID string, rule *NotificationRule) (*NotificationRule, *Response, error) {
+	lResp, _, lErr := s.ListNotificationRules(userID)
+	if lErr != nil {
+		return nil, nil, fmt.Errorf("[Channel Start delay must be unique for a given contact method]. Failed to fetch existing rules: %w", lErr)
+	}
+
+	for _, nr := range lResp.NotificationRules {
+		if isSameNotificationRule(nr, rule) {
+			return s.GetNotificationRule(userID, nr.ID)
+		}
+	}
+
+	return nil, nil, fmt.Errorf("user notification rule is already set with different attributes (possibly summary)")
```

Comment:
```suggestion
	return nil, nil, error)
```

Reply:
error needs to be passed in when calling s.searchExistingNotificationRule.  Updated to "[Channel Start delay must be unique for a given contact method]" here